src/core/uint.js

```diff
@@ -166,7 +186,7 @@ UInt.prototype.parse_generic = function(j) {
 };
 
 UInt.prototype.parse_hex = function(j) {
-  if (typeof j === 'string' && j.length === (this.constructor.width * 2)) {
+  if (new RegExp(`^[0-9A-F]{${this.constructor.width * 2}}$`).test(j)) {
```

Comment:
Well, ideally, rippled would reject lowercase or otherwise invalid InvoiceIDs, or InvoiceIDs that certainly won't match, but ripple-lib should also check, no?